test/bench/task.ts

```diff
@@ -0,0 +1,62 @@
+import { Suite } from './suite';
+import { convertToPerfSendFormat } from './util';
+import { performance } from 'perf_hooks';
+
+export class Task {
+  name: string;
+  parent: Suite;
+  data: unknown;
+  fn: (data: unknown) => void;
+  iterations: number;
+  transform?: (x: number) => number;
+  resultUnit: string;
+  args?: Record<string, string>;
+
+  constructor(
+    parent: Suite,
+    name: string,
+    data: unknown,
+    fn: (data: unknown) => void,
+    iterations: number,
+    resultUnit?: string,
+    transform?: (x: number) => number,
+    args?: Record<string, string>
+  ) {
+    this.parent = parent;
+    this.name = name;
+    this.iterations = iterations;
+    this.data = data;
+    this.fn = fn;
+    this.transform = transform;
+    this.args = args;
+    this.resultUnit = resultUnit ? resultUnit : 'ms';
+  }
+
+  run() {
+    console.log(`\t ${this.name} - iters: ${this.iterations}`);
+    // Warmup
+    for (let i = 0; i < this.iterations; i++) {
+      this.fn(this.data);
+    }
+    const results: number[] = [];
+
+    for (let i = 0; i < this.iterations; i++) {
+      const start = performance.now();
+      this.fn(this.data);
+      const end = performance.now();
```

Comment:
```suggestion
      const { data } = this
      const start = performance.now();
      this.fn(data);
      const end = performance.now();
```

One less step that is part of the measured code path. Probably want to pull data out at the top of run so the warm up is invoked the same way too